Update telemetry.md (#360)
mattermost/mattermost-plugin-incident-management: docs/telemetry.md

- `TeamID`: Unique identifier of the team where the incident channel is created
- `CreatedAt`: Timestamp of the incident creation
- `ChannelIDs`: A list containing a single element, the channel created along with the incident
- `PostID`: Unique identifier of the post
- `NumChecklists`: Number of checklists. It always equals 1 
- `TotalChecklistItems`: Number of checklist items this incident starts with. It always equals 0.
- `diagnosticID`: Unique identifier of the server the plugin is running on.
- `serverVersion`: Version of the server the plugin is running on.
- `pluginVersion`: Version of the plugin.
- `eventTimeStamp`: Timestamp indicating when the event was queued to send to the server.
- `event integrations`: Unused field. It always contains the value `null`.
- `event originalTimestamp`: Timestamp indicating when the event actually happened. It always equals `eventTimeStamp`.
- `type`: Type of the event. There are three event types that are tracked: `incident`, `tasks`, `playbook`.

For more information about telemetry, see [Mattermost Telemetry](https://docs.mattermost.com/administration/telemetry.html).

> ## Event data

Reviewer: I think we can delete everything below here

- `event originalTimestamp`: Timestamp indicating when the event actually happened. It always equals `eventTimeStamp`.
- `UserID`: Unique identifier of the server.
- `UserActualID`: Unique identifier of the user who initiated the action.
- `type`: Type of the event. There are three event types that are tracked: `incident`, `tasks`, `playbook`.

## Data collected in `incident` events

- `incidentID`: Unique identifier of the incident.
- `IsActive`: Boolean  value indicating if the incident is active.
- `CommanderUserID`: Unique identifier of the commander of the incident.
- `TeamID`: Unique identifier of the team where the incident channel is created.
- `CreatedAt`: Timestamp of the incident creation.
- `PostID`: Unique identifier of the post from which the incident was created (if relevant).
- `NumChecklists`: Number of stages in this incident.
- `TotalChecklistItems`: Number of tasks in this incident.
- `ActiveStage`: A number indicating the stage of the incident (0-based).
- `Public`: `true` if the incident was public, `false` if it was private.

## Data collected in `tasks` events

- `incidentID`: Unique identifier of the incident.

> - `NewState`: If the task was marked uncompleted. If the task was marked completed, the result will be `done`.

Reviewer: ```suggestion
- `NewState`: `null` if the task is uncompleted, `done` if the task was marked completed.
```

The following list details the types of metadata we collect:

## Data collected for all event types

- `serverVersion`: Version of the server the plugin is running on.
- `pluginVersion`: Version of the plugin.
- `eventTimeStamp`: Timestamp indicating when the event was queued to send to the server.
- `event originalTimestamp`: Timestamp indicating when the event actually happened. It always equals `eventTimeStamp`.
- `UserID`: Unique identifier of the server.
- `UserActualID`: Unique identifier of the user who initiated the action.
- `type`: Type of the event. There are three event types that are tracked: `incident`, `tasks`, `playbook`.

## Data collected in `incident` events

- `incidentID`: Unique identifier of the incident.
- `IsActive`: Boolean  value indicating if the incident is active.
- `CommanderUserID`: Unique identifier of the commander of the incident.

> - `TeamID`: Unique identifier of the team where the incident channel is created

Reviewer 2: ```suggestion
- `TeamID`: Unique identifier of the team where the incident channel is created.
```

- `CreatedAt`: Timestamp of the incident creation.
- `PostID`: Unique identifier of the post from which the incident was created (if relevant).
- `NumChecklists`: Number of stages in this incident.
- `TotalChecklistItems`: Number of tasks in this incident.
- `ActiveStage`: A number indicating the stage of the incident (0-based).

> - `Public`: True if the incident was public, false if it was private.

Reviewer 2: Some of the "true" and "false" words throughout the bullet points are formatted with an in-line code, others are not.

- `id`: Unique identifier of the event.
- `event integrations`: Unused field. It always contains the value `null`.
- `event originalTimestamp`: Timestamp indicating when the event actually happened. It always equals `eventTimeStamp`.
- `type`: Type of the event. There are three event types that are tracked: `incident`, `tasks`, `playbook`.

For more information about telemetry, see [Mattermost Telemetry](https://docs.mattermost.com/administration/telemetry.html).

> ## Incident data

Author: @cpoile, before I continue in this vein, I wanted to get your thoughts as to whether it's more scalable than the previous table. And also whether I've understood it correctly.

- `event originalTimestamp`: Timestamp indicating when the event actually happened. It always equals `eventTimeStamp`.

> - `type`: Type of the event. It always contains the string `track`.

Reviewer: this needs a bit of an update -- `type` is three types, and then `action` are those in the #359 PR notes
